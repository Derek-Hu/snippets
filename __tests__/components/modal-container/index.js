import React from 'react';
import ModalContainer from '~/components/modal-container/index';

describe('弹窗组件', () => {
  it('点击触发弹窗', () => {
    const onOk = jest.fn();
    const onCancel = jest.fn();

    const approveModalRef = React.createRef();
    const wrapper = mount(
      <ModalContainer
        title={'删除模板'}
        ref={approveModalRef}
        onOk={onOk}
        onCancel={onCancel}
        toggole={
          <span className={'triggerPoint'} style={{ color: 'red' }}>
            删除
          </span>
        }
      >
        <p>删除后不可恢复，确认要删除吗?</p>
      </ModalContainer>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().visible).toBe(false);
    // 点击删除按钮
    wrapper.find('span.triggerPoint').simulate('click');
    // 弹窗显示
    expect(wrapper.find('div.ant-modal-mask').hasClass('ant-modal-mask-hidden')).toBeFalsy();

    expect(wrapper.state().visible).toBe(true);

    approveModalRef.current.closeModal();
    expect(wrapper.state().visible).toBe(false);

    approveModalRef.current.showModal();
    expect(wrapper.state().visible).toBe(true);

    wrapper.find('div.ant-modal-content button.ant-btn.ant-btn-primary').simulate('click');

    approveModalRef.current.showModal();
    wrapper
      .find('div.ant-modal-footer button')
      .at(0)
      .simulate('click');

    expect(onOk).toBeCalled();
    expect(onCancel).toBeCalled();
  });

  it('确认函数出错时，弹窗隐藏', () => {
    const onOk = jest.fn(() => {
      throw new Error();
    });

    const wrapper = mount(
      <ModalContainer
        title={'删除模板'}
        onOk={onOk}
        toggole={
          <span className={'triggerPoint'} style={{ color: 'red' }}>
            删除
          </span>
        }
      >
        <p>删除后不可恢复，确认要删除吗?</p>
      </ModalContainer>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().visible).toBe(false);
    // 点击删除按钮
    wrapper.find('span.triggerPoint').simulate('click');
    // 弹窗显示
    expect(wrapper.find('div.ant-modal-mask').hasClass('ant-modal-mask-hidden')).toBeFalsy();

    wrapper.find('div.ant-modal-content button.ant-btn.ant-btn-primary').simulate('click');

    expect(onOk).toBeCalled();
  });

  it('无自定义函数，弹窗隐藏', () => {
    const wrapper = mount(
      <ModalContainer
        title={'删除模板'}
        toggole={
          <span className={'triggerPoint'} style={{ color: 'red' }}>
            删除
          </span>
        }
      >
        <p>删除后不可恢复，确认要删除吗?</p>
      </ModalContainer>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state().visible).toBe(false);
    // 点击删除按钮
    wrapper.find('span.triggerPoint').simulate('click');
    // 弹窗显示
    expect(wrapper.find('div.ant-modal-mask').hasClass('ant-modal-mask-hidden')).toBeFalsy();

    wrapper.find('div.ant-modal-content button.ant-btn.ant-btn-primary').simulate('click');

    expect(wrapper.state().visible).toBe(false);
  });
});
