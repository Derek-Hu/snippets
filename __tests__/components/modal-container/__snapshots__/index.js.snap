// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`弹窗组件 无自定义函数，弹窗隐藏 1`] = `
<ModalContainer
  title="删除模板"
  toggole={
    <span
      className="triggerPoint"
      style={
        Object {
          "color": "red",
        }
      }
    >
      删除
    </span>
  }
>
  <div
    onClick={[Function]}
  >
    <span
      className="triggerPoint"
      style={
        Object {
          "color": "red",
        }
      }
    >
      删除
    </span>
  </div>
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="删除模板"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <DialogWrap
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <Icon
            className="ant-modal-close-icon"
            type="close"
          />
        </span>
      }
      confirmLoading={false}
      footer={
        <LocaleReceiver
          componentName="Modal"
          defaultLocale={
            Object {
              "cancelText": "Cancel",
              "justOkText": "OK",
              "okText": "OK",
            }
          }
        >
          [Function]
        </LocaleReceiver>
      }
      forceRender={false}
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      onOk={[Function]}
      prefixCls="ant-modal"
      title="删除模板"
      transitionName="zoom"
      visible={false}
      width={520}
      wrapClassName=""
    />
  </Modal>
</ModalContainer>
`;

exports[`弹窗组件 点击触发弹窗 1`] = `
<ModalContainer
  onCancel={[MockFunction]}
  onOk={[MockFunction]}
  title="删除模板"
  toggole={
    <span
      className="triggerPoint"
      style={
        Object {
          "color": "red",
        }
      }
    >
      删除
    </span>
  }
>
  <div
    onClick={[Function]}
  >
    <span
      className="triggerPoint"
      style={
        Object {
          "color": "red",
        }
      }
    >
      删除
    </span>
  </div>
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="删除模板"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <DialogWrap
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <Icon
            className="ant-modal-close-icon"
            type="close"
          />
        </span>
      }
      confirmLoading={false}
      footer={
        <LocaleReceiver
          componentName="Modal"
          defaultLocale={
            Object {
              "cancelText": "Cancel",
              "justOkText": "OK",
              "okText": "OK",
            }
          }
        >
          [Function]
        </LocaleReceiver>
      }
      forceRender={false}
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      onOk={[Function]}
      prefixCls="ant-modal"
      title="删除模板"
      transitionName="zoom"
      visible={false}
      width={520}
      wrapClassName=""
    />
  </Modal>
</ModalContainer>
`;

exports[`弹窗组件 确认函数出错时，弹窗隐藏 1`] = `
<ModalContainer
  onOk={[MockFunction]}
  title="删除模板"
  toggole={
    <span
      className="triggerPoint"
      style={
        Object {
          "color": "red",
        }
      }
    >
      删除
    </span>
  }
>
  <div
    onClick={[Function]}
  >
    <span
      className="triggerPoint"
      style={
        Object {
          "color": "red",
        }
      }
    >
      删除
    </span>
  </div>
  <Modal
    confirmLoading={false}
    maskTransitionName="fade"
    okType="primary"
    onCancel={[Function]}
    onOk={[Function]}
    title="删除模板"
    transitionName="zoom"
    visible={false}
    width={520}
  >
    <DialogWrap
      closeIcon={
        <span
          className="ant-modal-close-x"
        >
          <Icon
            className="ant-modal-close-icon"
            type="close"
          />
        </span>
      }
      confirmLoading={false}
      footer={
        <LocaleReceiver
          componentName="Modal"
          defaultLocale={
            Object {
              "cancelText": "Cancel",
              "justOkText": "OK",
              "okText": "OK",
            }
          }
        >
          [Function]
        </LocaleReceiver>
      }
      forceRender={false}
      maskTransitionName="fade"
      okType="primary"
      onCancel={[Function]}
      onClose={[Function]}
      onOk={[Function]}
      prefixCls="ant-modal"
      title="删除模板"
      transitionName="zoom"
      visible={false}
      width={520}
      wrapClassName=""
    />
  </Modal>
</ModalContainer>
`;
